feat: add more troubleshooting examples (#2243)
shopware/docs: products/paas/shopware/guides/troubleshooting.md

---
nav:
  title: Troubleshooting Common Issues
  position: 90
---

# Troubleshooting Common Issues

This guide collects practical troubleshooting workflows for Shopware PaaS Native. It focuses on checks you can perform with the CLI, Grafana, and an application shell session.

## Prepare your CLI context

If you regularly work with the same organization, project, or environment, set a default context first:

```sh
sw-paas account context set
```

You can also preselect the default environment for the current shell session:

```sh
export SW_PAAS_ENVIRONMENT=staging
```

This avoids repeated prompts while you troubleshoot. For more details, see the [account guide](../fundamentals/account.md).

## Check timing and trace headers

If a page is slow, intermittently returns an error, or seems to be cached, inspect the current response timings first:

```sh
sw-paas app timing --query "foo=bar"
```

> Adding a query parameter helps bypass cached responses so you can inspect a fresh request.

Reviewer: ```suggestion
Adding a query parameter helps bypass cached responses, allowing you to inspect a fresh request.
```

The following response headers are especially useful:

| Header | Meaning |
| ------ | ------- |
| `x-envoy-upstream-service-time` | Time spent inside the Shopware PaaS Native infrastructure for the upstream request |
| `x-trace-id` | Trace identifier that you can search for in Grafana Tempo |
| `x-timer` | Fastly timing information. The `VE` value helps estimate the time between Fastly and the origin infrastructure |

You can also inspect the same headers in your browser's network tab.

## Check logs and traces in Grafana

Once you have confirmed the failing request, continue with logs and traces:

```sh
sw-paas open grafana
```

Use [logs](../monitoring/logs.md) to inspect application output and [traces](../monitoring/traces.md) to follow the request path through the system.

If you already captured an `x-trace-id`, search for that value directly in Tempo to jump to the affected request.

## My deployment finished, but the changes are not live

If an update appears to have completed but the application still serves the old state, inspect the application first:

```sh

> sw-paas application get --application-id <application-id>

Reviewer 2: Maybe we should also recommend the `sw-paas application deployment get` to see the different events that occurred during an deployment, e.g. an migration failure.

Reviewer 3: I agree, I think `sw-paas application deployment get` is more relevent than `sw-paas application get`.

Reviewer: @tombojer / @renaudhager - can you please update the most relevant command ?
